processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -257,4 +257,17 @@ public int hashCode()
 
     return Objects.hash(dataSource, descending, context, querySegmentSpec, duration, granularity);
   }
+
+  /**
+   * @param query
+   *
+   * @return intervals to query. For nested queries, return the intervals for the inner most query.
+   */
+  public static List<Interval> getIntervals(BaseQuery query)
```

Comment:
Since every query implementation extends `BaseQuery`, I think it's better to add this method to `Query` which is the higher interface than `BaseQuery`. What do you think about adding a method to `Query` like below:

```java
default List<Interval> getIntervalsOfInnerMostQuery()
  {
    if (getDataSource() instanceof QueryDataSource) {
      //noinspection unchecked
      return ((QueryDataSource) getDataSource()).getQuery().getIntervalsOfInnerMostQuery();
    } else {
      return getIntervals();
    }
  }
```